### PR TITLE
Dynamic sound loading

### PR DESCRIPTION
**Pull request by Phoenix-Starlight**
Implements dynamic sound (un)loading. Why? Because sound buffers are never unloaded and forever persist until exit or some trigger to release the buffers (reload).

### CODE REVIEW

File: common/src/main/java/org/embeddedt/modernfix/common/mixin/perf/dynamic_resources/SoundBufferLibraryMixin.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    private <K extends ResourceLocation, V extends CompletableFuture<SoundBuffer>> void onSoundRemoval(RemovalNotification<K, V> notification) {
+        if(!debugDynamicSoundLoading) {
+            notification.getValue().thenAccept(SoundBuffer::discardAlBuffer);
+            return;
+        }
+        notification.getValue().thenAccept(SoundBuffer::discardAlBuffer);
+        K k = notification.getKey();
+        if(k == null)
+            return;
+        ModernFix.LOGGER.warn("Evicted sound {}", k);
```
Comment (embeddedt) [inline common/src/main/java/org/embeddedt/modernfix/common/mixin/perf/dynamic_resources/SoundBufferLibraryMixin.java]: Any reason why this isn't tied to `debugDynamicSoundLoading` being enabled?

Comment (Phoenix-Starlight) [reply common/src/main/java/org/embeddedt/modernfix/common/mixin/perf/dynamic_resources/SoundBufferLibraryMixin.java]: It early returns but I suppose I can wrap it into an else. Refcatoring again.

Comment (embeddedt) [reply common/src/main/java/org/embeddedt/modernfix/common/mixin/perf/dynamic_resources/SoundBufferLibraryMixin.java]:
```suggestion
        if(debugDynamicSoundLoading)
            ModernFix.LOGGER.warn("Evicted sound {}", k);
```

Comment (Phoenix-Starlight) [reply common/src/main/java/org/embeddedt/modernfix/common/mixin/perf/dynamic_resources/SoundBufferLibraryMixin.java]: *facepalm* oops.

Comment (Phoenix-Starlight) [reply common/src/main/java/org/embeddedt/modernfix/common/mixin/perf/dynamic_resources/SoundBufferLibraryMixin.java]: Wait, no I can just wrap everything into the if except the sound close method.

Comment (Phoenix-Starlight) [reply common/src/main/java/org/embeddedt/modernfix/common/mixin/perf/dynamic_resources/SoundBufferLibraryMixin.java]: Force-pushed.

**Review by embeddedt — APPROVED** [review]
I'll do some testing before merging it but looks reasonable.

**Comment by embeddedt** [comment]
Had another thought: is `maximumSize` needed? Any new access to the cache will trigger removal of any expired entries, so we shouldn't need to cap the size ourselves.

**Comment by Phoenix-Starlight** [comment]
I was not aware that access would trigger removal, I was only
considering the pathological case where it tries to load more than 255 sounds (java channel limit).

**Pull request merged**